### PR TITLE
Formatting hours

### PR DESCRIPTION
**Pull request by shouston3**
Better formatting of hours slept

#388

### CODE REVIEW

File: web/views/sleep_tracker_view.ex [hunk 1]
```diff
@@ -22,10 +22,19 @@ defmodule Healthlocker.SleepTrackerView do
     if Kernel.length(past_week) == 0 do
       0
     else
-      total_slept / Kernel.length(past_week)
+      format_average_sleep(total_slept / Kernel.length(past_week))
     end
   end
 
+  # > format_average_sleep(7.375)
+  # > "7h 23m"
```
Comment (nelsonic) [inline web/views/sleep_tracker_view.ex]: comment does not reflect the format returned. should be `7h 23mins`
also, including this formatting function here means it doesn't have any tests. 🤔

**Comment by tonydaly** [comment]
Sorry, just saw this, but couldn't we use [Timex#format/2](https://hexdocs.pm/timex/Timex.Format.Duration.Formatter.html#format/2)?

This stuff usually has a bunch of edge cases which makes it difficult and time consuming to implement fully. For example in our implementation what if the number of minutes is 0. It looks like currently we'd show `5 hours and 0 minutes`, which might not be as useful?

**Comment by shouston3** [comment]
@tonydaly That seems like a much better option, we'll have a look at implementing that today, thanks

@nelsonic If not we'll ensure there are tests for this function (and the other functions in this file)

**Comment by shouston3** [comment]
@nelsonic Added a test using `@doc`

@tonydaly Now using timex, but I couldn't find a nicer format than [`:humanized`](https://hexdocs.pm/timex/1.0.0-pre/Timex.Format.Time.Formatters.Humanized.html) and then converting the string to the desired format with my own `hours_mins` function, let me know your thoughts on how I have implemented it

File: web/views/sleep_tracker_view.ex [hunk 2]
```diff
@@ -22,10 +22,35 @@ defmodule Healthlocker.SleepTrackerView do
...
+
+      iex> import Healthlocker.SleepTrackerView
+      iex> format_average_sleep(7.375)
+      "7h 22m"
+
+  """
+
+  def format_average_sleep(num) do
+    Duration.from_hours(num)
+      |> Timex.Format.Duration.Formatter.format(:humanized)
```
Comment (tonydaly) [inline web/views/sleep_tracker_view.ex]: 👍

File: web/views/sleep_tracker_view.ex [hunk 3]
```diff
@@ -22,10 +22,35 @@ defmodule Healthlocker.SleepTrackerView do
...
+      |> hours_mins()
```
Comment (tonydaly) [inline web/views/sleep_tracker_view.ex]: I _think_ we should be able to use the Timex Formatters for this, the especially [Strftime](https://hexdocs.pm/timex/Timex.Format.DateTime.Formatters.Strftime.html#content) one. We'll have to write our own one and pass it in as the formatter. There are some examples here: https://timex.readme.io/docs/formatting-dates

http://www.foragoodstrftime.com/ is a great resource to test your strftime format.

File: test/views/sleep_tracker_view_test.exs [hunk 4]
```diff
@@ -1,3 +1,4 @@
 defmodule Healthlocker.SleepTrackerViewTest do
   use Healthlocker.ConnCase, async: true
+  doctest Healthlocker.SleepTrackerView
```
Comment (tonydaly) [inline test/views/sleep_tracker_view_test.exs]: Nice

File: web/views/sleep_tracker_view.ex (hunk 3 above)
Comment (shouston3) [inline web/views/sleep_tracker_view.ex]: I found this a little tricky, but I'll check out the links after lunch and give it another go

Comment (shouston3) [inline web/views/sleep_tracker_view.ex]: @tonydaly Despite trying over the last couple of hours, I honestly haven't been able to make much of an improvement on this.
DateTimes work fine, but I don't think Durations work well with `strftime`

Comment (tonydaly) [inline web/views/sleep_tracker_view.ex]: Ahh that sucks, yeah looking at it the `Duration` formatters don't look as flexible 😞  We could maybe write our own, like [`Duration.Formatters.Humanized`](https://github.com/bitwalker/timex/blob/master/lib/format/duration/formatters/humanized.ex), but I don't think it'll be worth it.

**Pull request merged**